parser.go

```diff
@@ -1507,101 +1505,162 @@ func (p *parser) getTypeSpec(pkgPath, pkgName, typeName string) (*ast.TypeSpec,
 	return astTypeSpec, true
 }
 
-func (p *parser) parseSchemaPropertiesFromStructFields(pkgPath, pkgName string, structSchema *SchemaObject, astFields []*ast.Field) {
-	if astFields == nil {
-		return
-	}
-	structSchema.Properties = orderedmap.New()
-	if structSchema.DisabledFieldNames == nil {
-		structSchema.DisabledFieldNames = map[string]struct{}{}
+func (p *parser) parseAstFields(pkgPath, pkgName string, structSchema *SchemaObject, astFields []*ast.Field) {
+	for _, astField := range astFields {
+		p.parseAstField(pkgPath, pkgName, structSchema, astField)
 	}
+}
 
-	for _, astField := range astFields {
-		fieldSchema := &SchemaObject{}
-		typeAsString := p.getTypeAsString(astField.Type)
-		typeAsString = strings.TrimLeft(typeAsString, "*")
```

Comment:
from reviewing output of changes as part of this PR I found this was never used.

Reply:
Is this supposed to treat pointers as though they weren't pointers? Looks like `getTypeAsString` is unwrapping that for us anyway, so I'm guessing this is vestigial from a time before it was doing that? Good cleanup 👍

Reply:
yeah that's the same conclusion I came to